kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -454,6 +454,15 @@ private static class VacuumProtocolCheckTableFeature extends TableFeature.Reader
     }
   }
 
+  public static final TableFeature ALLOW_COLUMN_DEFAULTS_W_FEATURE =
+      new AllowColumnDefaultsTableFeature();
+
+  private static class AllowColumnDefaultsTableFeature extends TableFeature.WriterFeature {
```

Comment:
add a comment on the level of support: we only allow metadata updates and can't set or update default value columns